packages/next/client/experimental-script.tsx

```diff
@@ -83,13 +92,27 @@ const loadScript = (props: Props): void => {
   document.body.appendChild(el)
 }
 
-export default function Script(props: Props): JSX.Element | null {
+function handleClientScriptLoad(props: Props) {
+  if (props.strategy === 'defer') {
+    loadScript(props)
+  } else if (props.strategy === 'lazy') {
+    window.addEventListener('load', () => {
```

Comment:
What if the page already fired the `load` event before this code executes? Does that mean the script would never get loaded? AFAIK the `load` event could very well fire before JS gets a chance to execute. I may be wrong on this.

Reply:
That's a good point! I've added a fix